server/src/main/java/org/elasticsearch/transport/RemoteClusterPortSettings.java

```diff
@@ -0,0 +1,175 @@
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeValue;
+
+import java.util.List;
+import java.util.function.Function;
+
+import static org.elasticsearch.common.settings.Setting.boolSetting;
+import static org.elasticsearch.common.settings.Setting.intSetting;
+import static org.elasticsearch.common.settings.Setting.listSetting;
+import static org.elasticsearch.transport.TransportSettings.BIND_HOST_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.PORT_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.PUBLISH_HOST_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.PUBLISH_PORT_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_KEEP_ALIVE_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_KEEP_COUNT_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_KEEP_IDLE_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_KEEP_INTERVAL_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_NO_DELAY_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_RECEIVE_BUFFER_SIZE_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_REUSE_ADDRESS_PROFILE;
+import static org.elasticsearch.transport.TransportSettings.TCP_SEND_BUFFER_SIZE_PROFILE;
+
+/**
+ * Contains the settings and some associated logic for the settings related to the Remote Access port, used by Remote Cluster Security 2.0.
+ */
+public class RemoteClusterPortSettings {
+    private static final Logger logger = LogManager.getLogger(RemoteClusterPortSettings.class);
+
+    public static final String REMOTE_CLUSTER_PROFILE = "_remote_cluster";
+    public static final String REMOTE_CLUSTER_PREFIX = "remote_cluster.";
+    public static final String REMOTE_CLUSTER_SSL_PREFIX = REMOTE_CLUSTER_PREFIX + "ssl.";
+
+    public static final Setting<Boolean> REMOTE_CLUSTER_PORT_ENABLED = boolSetting(
+        REMOTE_CLUSTER_PREFIX + "enabled",
+        false,
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<List<String>> HOST = listSetting(
+        REMOTE_CLUSTER_PREFIX + "host",
+        TransportSettings.BIND_HOST,
+        Function.identity(),
+        Setting.Property.NodeScope
+    );
+    public static final Setting<List<String>> PUBLISH_HOST = listSetting(
+        REMOTE_CLUSTER_PREFIX + "publish_host",
+        HOST,
+        Function.identity(),
+        Setting.Property.NodeScope
+    );
+    public static final Setting<List<String>> BIND_HOST = listSetting(
+        REMOTE_CLUSTER_PREFIX + "bind_host",
+        HOST,
+        Function.identity(),
+        Setting.Property.NodeScope
+    );
+
+    public static final Setting<Integer> PORT = intSetting(REMOTE_CLUSTER_PREFIX + "port", 9900, 0, 65535, Setting.Property.NodeScope);
```

Comment:
I noticed this too. Port 0 lets the OS pick a port number dynamically. That is OK for tests, but customers will probably want to use a static port in production.

Potentially, we might consider a separate PR to log a warning when 0 is configured, but I don't think it is specific to RCS 2.0. I think it would apply to all ports and all protocols.

Reply:
Yeah, I don't think there's much we can do about this in this PR. We do have a few settings where we allow values you probably wouldn't want in production for testing purposes, which may be a problem we should address more generally.